### ISSUE TITLE
Item Preferences Still Very Broken

### ISSUE THREAD

**Issue by Delegator2** [WarEmu/WarBugs#18360]
The patch this week said that item preferences in inventory were now working, however I have observed several issues:

1. Items filtered to bags sometimes are not visible in inventory, looking like blank spots even though they are there. If you select the slot and move to another, the item appears, but this is very disconcerting
2. Items that should be filtered to the crafting bag are not properly moved there if there are no slots in the primary inventory bags that would take them. For example, let's say you designate one bag for crafting items, but the bag is full. Trying to get mail that has crafting items will have problems because the items are not correctly put in the crafting bag due to the primary inventory bag being full
3. I have had items simply disappear when getting them from mail, even though they should be filtered to a specific bag

In short: great idea, would be super useful, but still not completely working.

**Comment by CrimsonFenris** [comment]
Noted several issues as well :

4. Adding item which stacks into existing ones will automatically move the item into assigned bag (for example potions).
5. Receiving existing currencies will often results into switching of currencies order in bag, messing up our manual sorting despite currency bags dont have sorting options. The issues happen as soon any filtering option is checked, and seems to disappear when filtering is completely disabled for all bags.

**Comment by Tyrannicus** [comment]
Since the January 15-16 patch I have invisible items in my backpack. I constantly receive the "could not move item" error messages when opening mail, when looting, when crafting, when receiving medallions after a roll and when opening bags. The item will be invisible in my backpack until i /reload or click/move the item. 

This issue happened on live server some time after launch.  If i recall correctly a gm had to do something with my account in game in order to reset the backpack so items were no longer invisible though I could be mistaken.

**Comment by lilimx** [comment]
related/similar to #18114
